### ISSUE TITLE
Importing an unknown member of a module causes all types in a stub file to become potentially Unknown

### ISSUE THREAD

**Issue by Avasam** [microsoft/pyright#3328]
**Describe the bug**
In a `.pyi` file, if using an import that would cause the error `is not a known member of module`, all of the types and generic parameters will become potentially `Unknown`. ie `SomeClass` becomes `SomeClass | Unknown`

**To Reproduce**
https://github.com/pywinrt/python-winsdk/blob/f1154f4d0cb2216809d36ddc8aca61a120f3f6cc/pywinrt/winsdk/windows/graphics/capture/__init__.pyi
At line 47, `import winsdk.windows.ui.composition` could not be resolved.
Then it is used at line 90 `def create_from_visual(visual: typing.Optional[winsdk.windows.ui.composition.Visual]) -> typing.Optional[GraphicsCaptureItem]: ...`
![image](https://user-images.githubusercontent.com/1350584/162600745-bec00347-07e3-49cd-829a-761e2cb371b4.png)
See this issue for more details: https://github.com/pywinrt/python-winsdk/issues/8

> This happens with all `_ns_module.*` types as far as I can tell, but here's an example from `windows.graphics.directx.direct3d11.__init__.py`:
> ![image](https://user-images.githubusercontent.com/1350584/162583110-d7470106-c48f-4618-b507-d4e074073773.png)
> 
> Then in `windows.graphics.capture.__init__.pyi`
> ![image](https://user-images.githubusercontent.com/1350584/162583141-37df84dc-aef8-4993-bf42-8622d0ac30fa.png)
> 
> Which produces strict type-checking errors in my code, unless I explicitly specify the type:
> ![image](https://user-images.githubusercontent.com/1350584/162583253-481c71f0-2800-4e5f-af70-ad922dee4310.png)
> 
> Even though `IDirect3DSurface` is defined in `windows.graphics.directx.direct3d11.__init__.pyi`:
> ![image](https://user-images.githubusercontent.com/1350584/162583476-3f23719f-8d4b-45cd-baff-2056bd9bd3d9.png)

**Expected behavior**
Only the value that tries to use the missing import should be `Unknown`, not everything in the type stub

**Workaround**
Either aliasing the import, or using a partial import path.
> I can confirm that changing the import to `import winsdk.windows.ui.composition as _winsdk_windows_ui_composition` in the pyi file only and ` def create_from_visual(visual: _winsdk_windows_ui_composition.Visual) -> GraphicsCaptureItem: ...` Does work as a fix
> ![image](https://user-images.githubusercontent.com/1350584/162600494-374f2fa7-ff2b-477a-ad7c-c9844ef7b3f4.png)
> 
> Even `from winsdk.windows.ui import composition` then `composition.Visual` seems to work (not sure if there's other side effects). I guess as long as you don't use `winsdk.windows.ui.composition.<something>` directly in the type definition.
> 
> Causes issues:
> ![image](https://user-images.githubusercontent.com/1350584/162600745-bec00347-07e3-49cd-829a-761e2cb371b4.png)
> 
> Ok?:
> ![image](https://user-images.githubusercontent.com/1350584/162600769-0f27f0b3-0e29-406c-a683-045cfad356c2.png)

**VS Code extension or command-line**
Pylance

**Comment by erictraut** [comment]
I'm confused by the bug report. An unresolved import within a type stub file should not cause other unrelated symbols to be evaluated as `Unknown`. If the unresolved symbol is used within another type (e.g. as a base class or within a type annotation), that symbol will naturally have an `Unknown` type.

I tried installing `winsdk` from pip, but it apparently doesn't install on non-Windows machines.

I don't know what `_ns_module.*` refers to, so I wasn't able to follow the problem in the linked bug report.

Perhaps you could create a simplified example that demonstrates the problem you're seeing?

**Comment by Avasam** [comment]
Hi Eric. Looking further into it, I think I do have a small repro I can do. Here's an example with enum (any module would work)
```
import enum
import enum.does_not_exists


class MyClass(enum.Enum):
    ...
```
![image](https://user-images.githubusercontent.com/1350584/162666317-d168e7a0-5742-46f8-ad41-488ddedbb4e2.png)


Trying to import the invalid `enum.does_not_exists` is sufficient for all `enum.*` to be seen as `Unknown`. Which replicates the behaviour seen in the python-winsdk library:
![image](https://user-images.githubusercontent.com/1350584/162666663-de5e6920-a4b2-4724-b1b2-c603e3f980ae.png)


Simply adding an alias to the invalid import is a workaround:
![image](https://user-images.githubusercontent.com/1350584/162666562-17fd0145-1c2a-4bec-a3c6-63af369aff1d.png)

**Comment by erictraut** [comment]
Thanks, that's a great repro. I was able to find the root cause and fix it. This bug has existed in pyright for at least two years. I'm surprised that no one has reported it previously!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.237, which I just published. It will also be included in the next release of pylance.
